Anonymous User Lesson Overlay

Creeland opened the pull request:
This new overlay is served to anonymous users when the complete a lesson. It gives them:
- The option to continue, re-watch, or view the full course
- The ability to search from the lesson overlay
- A recommendation for the primary tag of the lesson's curated page
- Links to two popular free courses from the lesson's primary tag

<img width="1511" alt="image" src="https://github.com/skillrecordings/egghead-next/assets/518406/e3aaedb7-e22f-49c8-a18c-880ab418b63e">
<img width="1511" alt="image" src="https://github.com/skillrecordings/egghead-next/assets/518406/643bfd7f-55f6-4cd9-8be9-5b4b0e14df9f">
<img width="1508" alt="image" src="https://github.com/skillrecordings/egghead-next/assets/518406/eadbd468-2c48-4c59-ad67-e4e074d3a917">
<img width="756" alt="image" src="https://github.com/skillrecordings/egghead-next/assets/518406/b65ac984-0b14-4a98-97e3-1f12696635d4">

vojtaholik commented on src/components/pages/lessons/overlay/anon-user-overlay.tsx:
```diff
@@ -0,0 +1,318 @@
+        </div>
+        <div className="flex flex-col items-center gap-y-2 3xl:mt-4">
+          {lesson?.collection?.path && (
+            <Link href={lesson.collection.path}>
+              <a
+                onClick={() => {
+                  track('clicked view course', {
+                    lesson: lesson.slug,
+                    location: 'lesson overlay',
+                    feature: FEATURE,
+                  })
+                }}
+                className="w-full"
+              >
+                <button className="bg-blue-600 rounded py-2 flex w-full items-center justify-center hover:bg-blue-500 transition-colors duration-200 ease-in-out text-xs md:text-base whitespace-nowrap">
+                  <AcademicCapIcon className="w-6 mr-2" />
+                  Watch Full Course
+                </button>
+              </a>
+            </Link>
```
nesting a button inside link is not allowed

vojtaholik commented on src/components/pages/lessons/overlay/anon-user-overlay.tsx:
```diff
@@ -0,0 +1,318 @@
+        </div>
+        <div className="flex flex-col items-center gap-y-2 3xl:mt-4">
+          {lesson?.collection?.path && (
+            <Link href={lesson.collection.path}>
```
I honestly wouldn't expect for "Watch Full Course" to take me to course landing page.

Creeland commented:
Thanks for the feedback on the course link @vojtaholik. I styled the `a` tag, removed the `button` element, and changed the link text to `View Course`
